ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java

```diff
@@ -1030,6 +1039,271 @@ private Pair<ServerErrorCode, GetResponse> getBlobFromRemoteReplica(ReplicateBlo
     }
   }
 
+  // LOCAL_CONSISTENCY_TODO handleReplicateBlobRequestV2 will replace handleReplicateBlobRequest totally.
+  // The deployment procedure will be
+  // 1. We'll deploy the server code everywhere
+  // 2. Switch frontend from v1 to v2 to test it in Perf and EI
+  // 3. Switch frontend from v1 to v2 in PROD
+  // 4. Remove old code handleReplicateBlobRequest code which supports v1 in server
+  public void handleReplicateBlobRequestV2(NetworkRequest request, ReplicateBlobRequest replicateBlobRequest)
+      throws IOException, InterruptedException {
+    long startProcessTime = SystemTime.getInstance().milliseconds();
+    // Get the parameters from the replicateBlobRequest: the blobId, the remoteHostName and the remoteHostPort.
+    BlobId blobId = replicateBlobRequest.getBlobId();
+    String remoteHostName = replicateBlobRequest.getSourceHostName();
+    int remoteHostPort = replicateBlobRequest.getSourceHostPort();
+
+    // check the request parameters.
+    // operationType should be either TtlUpdateRequest or DeleteRequest to fix TtlUpdate record or delete record.
+    if (replicateBlobRequest.getOperationType() != RequestOrResponseType.TtlUpdateRequest
+        && replicateBlobRequest.getOperationType() != RequestOrResponseType.DeleteRequest) {
+      logger.error("ReplicateBlobRequest invalid request {}", replicateBlobRequest);
+      completeReplicateRequest(request, replicateBlobRequest, ServerErrorCode.Bad_Request, startProcessTime);
+      return;
+    }
+    // LOCAL_CONSISTENCY_TODO
+    // For lifeVersion, suppose we should always provide MessageInfo.LIFE_VERSION_FROM_FRONTEND. Any other cases?
+    if (replicateBlobRequest.getLifeVersion() != MessageInfo.LIFE_VERSION_FROM_FRONTEND) {
+      logger.info("ReplicateBlobRequest invalid life version {}", replicateBlobRequest);
+      completeReplicateRequest(request, replicateBlobRequest, ServerErrorCode.Bad_Request, startProcessTime);
+      return;
+    }
```

Comment:
nit: can we just merge these two if statement, we probably don't need to differentiate what error it is in the log message, we should be able to find if it's bad request due to operation type of the life version.